internal/gqlclient/gqlclient.go

```diff
@@ -371,5 +372,20 @@ func (c *Client) runWithFiles(ctx context.Context, gqlReq RequestWithFiles, resp
 	if err := json.Unmarshal(resp, &gr); err != nil {
 		return errors.Wrap(err, "decoding response")
 	}
-	return nil
+
+	if len(intermediateResp) == 1 {
```

Comment:
Nit: a comment describing why this conditional is here or what conditions it may occur under would probably be useful for anyone debugging around this area in the future.